solidity/contracts/AugurConstantProductMarket.sol

```diff
@@ -148,6 +146,24 @@ contract AugurConstantProduct is ERC20 {
 		shareToken.unsafeBatchTransferFrom(from, to, tokenIds, amounts);
 	}
 
+	// formula where x = yes shares owned when swapping yes
+	// s = (nf + 1000x -xf - 1000y - 1000n + sqrt(x^2f^2 - 2000x^2f + 1000000x^2 - 2000000nx - 2nxf^2 + 4000nxf + 2000000xy - 2000xfy + 1000000n^2 + n^2f^2 - 2000n^2f + 1000000y^2 + 2000000ny - 2000nfy)) / 2(1000 - f)
```

Comment:
It may be worthwhile to include details in this comment about where you got this from.  I don't think we need a full derivation, but perhaps just something that indicated what this is derived from would be useful I think.